docs/reference/contents.md

```diff
@@ -0,0 +1,129 @@
+# Contents
+
+This section describes how is the contents integrated in the website and how it is used.
+
+## Access within the kernel
+
+By default the contents through the filebrowser is independent of the contents
+accessible within an execution kernel. Sharing the contents with the kernel is dependent
+of that kernels.
+
+### Emscripten kernel
+
+Kernels using Emscripten (like [pyodide](https://github.com/jupyterlite/pyodide-kernel/)
+or [xeus kernels](https://github.com/jupyterlite/xeus/)) relies on the
+[Emscripten filesystem](https://emscripten.org/docs/api_reference/Filesystem-API.html)
+to access their contents. For such case, `@jupyterlite/contents` provides a helper class
+[`DriveFS`](https://jupyterlite.readthedocs.io/en/stable/reference/api/ts/classes/jupyterlite_contents.DriveFS-1.html)
+to be mounted within Emscripten filesystem with a snippet like:
+
+```ts
+  const mountpoint = '/drive';
+  const { FS, PATH, ERRNO_CODES } = /* provided by the empscripten module */;
+  const { baseUrl } = options;
+  const { DriveFS } = await import('@jupyterlite/contents');
+
+  const driveFS = new DriveFS({
+    FS,
+    PATH,
+    ERRNO_CODES,
+    baseUrl, // Website base URL
+    driveName: 'my-drive', // Any name of your choosing
+    mountpoint,
+  });
+  FS.mkdir(mountpoint);
+  FS.mount(driveFS, {}, mountpoint);
+  FS.chdir(mountpoint);
+```
+
+After mounting the drive, the Jupyter Server contents (the one displayed in the
+filebrowser) will be available within the kernel under the folder `/drive`.
+
+The website base URL is required for the `DriveFS` to request its content from the main
+application. Diving into the drive architecture will clarify that.
+
+```{mermaid}
+flowchart LR
+    subgraph main thread
+        direction TB
+        M[Main thread]
+        M --- C[Contents]
+    end
+    subgraph webworker
+        direction TB
+        M -.- K[Kernel]
+        K --- FS[Emscripten FS]
+        FS --- D[DriveFS]
+    end
+    S[Service worker] -.-|BroadcastChannel| M
+    D -.-|REST API| S
+```
+
+Three threads are at play when running a kernel inside JupyterLite:
+
+- The main thread: it executes the main user interface and knows about the filebrowser
+  contents.
+- The kernel web worker: it executes the kernel (e.g. evaluate the code snippet from a
+  notebook sent by the main thread). It mounts a `DriveFS` into the Emscripten
+  filesystem.
+- The service worker: it serves website assets from cache (to work offline). And it also
+  can capture any other network requests.
```

Comment:
```suggestion
- The service worker: it serves website assets from cache (to work offline). And it can also
  capture any other network requests.
```